### PR TITLE
chore: split Arbitrum Reader and DB modules for improved maintainability

### PR DESCRIPTION
**Pull request by akolotov**
**This PR compliments the changes made under https://github.com/blockscout/blockscout/pull/11485, that is why is: the merge order #11531 first, then #11485**

## Motivation

The current implementation has large monolithic files (`reader.ex` and `db.ex`) that are difficult to maintain and modify, especially when using LLM-based coding assistants. This PR refactors these files into smaller, focused modules to improve code organization and maintainability while optimizing for LLM-based development workflows.

### Enhancements

- Split `Explorer.Chain.Arbitrum.Reader` into specialized modules:
  - `api.ex` - API endpoint-specific functions
  - `common.ex` - Shared query functionality
  - `indexer/*.ex` - Specialized indexer functions
- Refactored `Indexer.Fetcher.Arbitrum.Utils.Db` into domain-specific modules:
  - `common.ex` - Chain-agnostic database operations
  - `messages.ex` - Cross-chain message handling
  - `parent_chain_transactions.ex` - L1 transaction management
  - `settlement.ex` - Batch and confirmation data
  - `tools.ex` - Internal helper functions
- Centralized Arbitrum-specific constants:
  - Created new module `EthereumJSONRPC.Arbitrum.Constants.Events`:
    - Consolidated event signatures and parameter types
  - Created new module `EthereumJSONRPC.Arbitrum.Constants.Contracts`:
    - Consolidated contract ABIs, function selectors, and addresses
- Added comprehensive documentation for each new module
- Added README.md explaining the module organization and usage guidelines

## Checklist for your Pull Request (PR)

- [ ] If I added new functionality, I added tests covering it.
- [ ] If I fixed a bug, I added a regression test to prevent the bug from silently reappearing again.
- [ ] I checked whether I should update the docs and did so by submitting a PR to [docs repository](https://github.com/blockscout/docs).
- [ ] If I added/changed/removed ENV var, I submitted a PR to [docs repository](https://github.com/blockscout/docs) to update the list of [env vars](https://github.com/blockscout/docs/blob/master/setup/env-variables/README.md) and I updated the version to `master` in the Version column. If I removed variable, I added it to [Deprecated ENV Variables](https://github.com/blockscout/docs/blob/master/setup/env-variables/deprecated-env-variables/README.md) page. After merging docs PR, changes will be reflected in these [pages](https://docs.blockscout.com/setup/env-variables).
- [ ] If I added new DB indices, I checked, that they are not redundant, with PGHero or other tools.
- [ ] If I added/removed chain type, I modified the Github CI matrix and PR labels accordingly.

### CODE REVIEW

**Review by vbaranov — COMMENTED** [review]
@akolotov I merged #11541 to the master branch. Please rebase PR to it - I still see changes from #11541 in the current PR.

**Comment by akolotov** [comment]
> @akolotov I merged #11541 to the master branch. Please rebase PR to it - I still see changes from #11541 in the current PR.

@vbaranov you can use the following link to review the branch state before the most recent commit (which is the merge of  #11541 in `chore/arbitrum/modularise-reader`): https://github.com/blockscout/blockscout/pull/11531/files/5cca05c4387405407a27796287afc30614799b80

File: apps/indexer/lib/indexer/fetcher/arbitrum/utils/db/messages.ex [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+
+  alias Explorer.Chain.Block, as: FullBlock
+  alias Explorer.Chain.Hash
+
+  alias Indexer.Fetcher.Arbitrum.Utils.Db.Tools, as: DbTools
+
+  require Logger
+
+  # 32-byte signature of the event L2ToL1Tx(address caller, address indexed destination, uint256 indexed hash, uint256 indexed position, uint256 arbBlockNum, uint256 ethBlockNum, uint256 timestamp, uint256 callvalue, bytes data)
+  @l2_to_l1_event "0x3e7aafa77dbf186b7fd488006beff893744caa3c4f6f299e8a709fa2087374fc"
```
[the remark this answers is not in the document]
Comment (akolotov) [reply apps/indexer/lib/indexer/fetcher/arbitrum/utils/db/messages.ex]: Introduced a separate module for Arbitrum related events in a9c8b8b

[the remark this answers is not in the document]
Comment (akolotov) [reply apps/indexer/lib/indexer/fetcher/arbitrum/utils/db/messages.ex]: I also moved all constants (selectors, ABI) related to the Arbitrum contracts to under e547688

File: apps/indexer/lib/indexer/fetcher/arbitrum/utils/db/settlement.ex [hunk 2]
```diff
@@ -0,0 +1,482 @@
...
+  def get_batch_by_rollup_block_number(num)
+      when is_integer(num) and num >= 0 do
+    case Reader.get_batch_by_rollup_block_number(num) do
+      nil ->
+        nil
+
+      batch ->
+        case batch.commitment_transaction do
+          nil ->
+            raise "Incorrect state of the DB: commitment_transaction is not loaded for the batch with number #{num}"
```
[the remark this answers is not in the document]
Comment (akolotov) [reply apps/indexer/lib/indexer/fetcher/arbitrum/utils/db/settlement.ex]: addressed in a7f71e6

File: apps/indexer/lib/indexer/fetcher/arbitrum/utils/db/settlement.ex [hunk 3]
```diff
@@ -0,0 +1,482 @@
...
+
+    ## Returns
+    - The next L1 block number after the latest committed batch or `value_if_nil` if no committed batches are found.
+  """
+  @spec l1_block_to_discover_latest_committed_batch(FullBlock.block_number() | nil) :: FullBlock.block_number() | nil
+  def l1_block_to_discover_latest_committed_batch(value_if_nil)
+      when (is_integer(value_if_nil) and value_if_nil >= 0) or is_nil(value_if_nil) do
+    case Reader.l1_block_of_latest_committed_batch() do
+      nil ->
+        log_warning("No committed batches found in DB")
```
[the remark this answers is not in the document]
Comment (akolotov) [reply apps/indexer/lib/indexer/fetcher/arbitrum/utils/db/settlement.ex]: addressed in a7f71e6

File: apps/indexer/lib/indexer/fetcher/arbitrum/utils/db/messages.ex [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+    ## Returns
+    - The L1 block number following the block that contains the latest execution transaction
+      for an L2-to-L1 message, or `value_if_nil` if no such executions have been found.
+  """
+  @spec l1_block_to_discover_latest_execution(nil | FullBlock.block_number()) :: nil | FullBlock.block_number()
+  def l1_block_to_discover_latest_execution(value_if_nil)
+      when (is_integer(value_if_nil) and value_if_nil >= 0) or is_nil(value_if_nil) do
+    case Reader.l1_block_of_latest_execution() do
+      nil ->
+        log_warning("No L1 executions found in DB")
```
[the remark this answers is not in the document]
Comment (akolotov) [reply apps/indexer/lib/indexer/fetcher/arbitrum/utils/db/messages.ex]: addressed in a7f71e6

File: apps/indexer/lib/indexer/fetcher/arbitrum/utils/db/messages.ex [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+    ## Returns
+    - The L1 block number immediately following the latest discovered message to L2,
+      or `value_if_nil` if no messages to L2 have been found.
+  """
+  @spec l1_block_to_discover_latest_message_to_l2(nil | FullBlock.block_number()) :: nil | FullBlock.block_number()
+  def l1_block_to_discover_latest_message_to_l2(value_if_nil)
+      when (is_integer(value_if_nil) and value_if_nil >= 0) or is_nil(value_if_nil) do
+    case Reader.l1_block_of_latest_discovered_message_to_l2() do
+      nil ->
+        log_warning("No messages to L2 found in DB")
```
[the remark this answers is not in the document]
Comment (akolotov) [reply apps/indexer/lib/indexer/fetcher/arbitrum/utils/db/messages.ex]: addressed in a7f71e6

File: apps/indexer/lib/indexer/fetcher/arbitrum/utils/db/messages.ex (hunk 1 above)
[the remark this answers is not in the document]
Comment (vbaranov) [reply apps/indexer/lib/indexer/fetcher/arbitrum/utils/db/messages.ex]: Looks great!

**Review by vbaranov — APPROVED** [review]

**Pull request merged**